service/resource/configmap/resource.go

```diff
@@ -252,19 +261,19 @@ func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desire
 		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d config maps that have to be updated", len(configMapsToUpdate)))
 	}
 
-	return configMapsToCreate, configMapsToDelete, configMapsToUpdate, nil
+	return configMapsToUpdate, nil
 }
 
 func (r *Resource) Name() string {
 	return Name
 }
 
-func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState interface{}) error {
+func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
```

Comment:
I would like to move to separate files in a separate PR. I did not do it here to keep the diff as clean as possible.